src/oidcop/oauth2/authorization.py

```diff
@@ -429,11 +429,16 @@ def pick_authn_method(self, request, redirect_uri, acr=None, **kwargs):
         if auth_id:
             return _context.authn_broker[auth_id]
 
+        res = None
         if acr:
             res = _context.authn_broker.pick(acr)
         else:
-            res = pick_auth(_context, request)
-
+            try:
+                res = pick_auth(_context, request)
+            except Exception as e:
+                logger.debug(
+                    f"An error occurred while picking the authN broker: {exc}"
```

Comment:
Can an exception be raised here?

Reply:
Really don't know, I Just adapt the old code and take It as It was but moving in different context

@rohe do we have tests for this? I think so

Reply:
Ah, typo error, e and not ecco!

Reply:
paradoxically we are sure that the current tests do not fail, otherwise the typo error would have created a higher order exception

Reply:
fixed: https://github.com/IdentityPython/oidc-op/pull/66/commits/c69fbfcd0a93a95d79ba5bd9a0327731c8a920dd

Reply:
@nsklikas does it sounds good to you or should we still improve this?

Reply:
Seems to be covered by the tests.